src/v/raft/consensus.cc

```diff
@@ -2020,6 +2025,8 @@ consensus::do_append_entries(append_entries_request&& r) {
               "present, request: {}, current state: {}",
               request_metadata,
               meta());
+            // force current leader step down
+            reply.force_stepdown = true;
```

Comment:
Why should the leader step down? Previously I was thinking that this branch was for outdated reordered requests but looks like now we can hit it if (the visible part of) the follower log is for real longer than the leader log? I wonder what is then preventing this leader from getting elected again and repeating the issue...

If this is the case then we should probably change the election rules as well instead of forcing stepdown.

Reply:
i think you are right, i will drop the stepdown part, the change is preventing from truncations being a result of redeliveries, stepdown is not required here.